### PR TITLE
add dependbot

### LINKED ISSUE #1472: Keep all datafusion's packages up to date with Dependabot
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**
As the title.

**Describe the solution you'd like**
Add a `dependabot.yml` into `.github`. 

https://github.com/marketplace/actions/dependencies-autoupdate
https://github.blog/2020-06-01-keep-all-your-packages-up-to-date-with-dependabot/

**Describe alternatives you've considered**
No

**Additional context**
Suitable for newcomers.

### PR DESCRIPTION
**Pull request by xudong963**
# Which issue does this PR close?

Closes #1472 

 # Rationale for this change

# What changes are included in this PR?

ref: https://github.blog/2020-06-01-keep-all-your-packages-up-to-date-with-dependabot/
# Are there any user-facing changes?

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+updates:
+  - package-ecosystem: cargo
+    directory: "/"
+    schedule:
+      interval: weekly
+      day: sunday
+      time: "7:00"
+    open-pull-requests-limit: 10
+    target-branch: master
+    labels: [auto-dependencies]
```
Comment (xudong963) [inline .github/dependabot.yml:11]: I think we should add a new label for tickets about auto-dependency

Comment (alamb) [reply .github/dependabot.yml:11]: Created: https://github.com/apache/arrow-datafusion/labels/auto-dependencies

**Review by alamb — APPROVED** [review]
Looks great @xudong963  -- I do wonder if we also need to ensure that the security settings are updated correctly for dependabot as described in https://github.blog/2020-06-01-keep-all-your-packages-up-to-date-with-dependabot/

Screen Shot 2021-12-28 at 6 37 23 AM

I poked around other apache repos, and my results are inconclusive -- there are some repos where it is working nicely (e.g https://github.com/apache/iotdb/pull/4555) and others e.g. https://issues.apache.org/jira/browse/INFRA-21098 where apparently something else is needed.

I think we should merge this one in and then debug it if it isn't working

Thanks @xudong963

**Comment by xudong963** [comment]
> I do wonder if we also need to ensure that the security settings are updated correctly for dependabot as described in https://github.blog/2020-06-01-keep-all-your-packages-up-to-date-with-dependabot/

I think it'll be better to enable it. (BTW, I don't have access.

**Comment by alamb** [comment]
> I think it'll be better to enable it. (BTW, I don't have access.

Yeah, I don't have access either -- if this doesn't work we'll have to file a request in the Apache Infrastructure JIRA to make it happen. However, it is fascinating that IOTDB seems to have it working without a corresponding infra request, so we'll see

**Comment by xudong963** [comment]
Unfortunately, it didn't work

**Comment by houqp** [comment]
@xudong963 you can take a look at https://cwiki.apache.org/confluence/display/INFRA/Git+-+.asf.yaml+features to see if this option can be configured in a self-serviced way.

**Comment by alamb** [comment]
@xudong963  I wonder if it might be related to https://github.com/dependabot/dependabot-core/issues/4574 as well

**Pull request merged**